### PR TITLE
Feature/plcn 329 create mail options

### PR DESCRIPTION
**Pull request by AntonEsbjergIT-Minds**
A service that can send emails has been created. This service uses SMTP from SendGrid and the email is sent using a mail setup to be used by SendGrid. The Email is composed from a razor template and the content is set by an admin in the database.

### CODE REVIEW

**Review by EsBjerrum — APPROVED** [review]

File: src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailButton.cshtml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+<div align="center" class="button-container" style="padding-top:15px;padding-right:10px;padding-bottom:15px;padding-left:10px;">
+    <!--[if mso]>
+      <table width="100%" cellpadding="0" cellspacing="0" border="0" style="border-spacing: 0; border-collapse: collapse; mso-table-lspace:0pt; mso-table-rspace:0pt;">
+        <tr>
+          <td style="padding-top: 15px; padding-right: 10px; padding-bottom: 15px; padding-left: 10px" align="center">
+            <v:roundrect xmlns:v="urn:schemas-microsoft-com:vml" xmlns:w="urn:schemas-microsoft-com:office:word" href="@Model.Url" style="height:27.75pt;width:132.75pt;v-text-anchor:middle;" arcsize="65%" strokeweight="0.75pt" strokecolor="#6A971F" fillcolor="#6a971f">
+              <w:anchorlock/>
+              <v:textbox inset="0,0,0,0">
+                <center style="color:#ffffff; font-family:Arial, sans-serif; font-size:14px">
+                  <![endif]-->
```
Comment (itm-mne) [inline src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailButton.cshtml:14]: Maybe add TODO, if this section is gonna be used later.

File: src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailButton.cshtml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            </v:textbox>
+          </v:roundrect>
+        </td>
+      </tr>
+      <tr>
+        <td>
+        </td>
+      </tr>
+    </table>
+    <![endif]-->
```
Comment (itm-mne) [inline src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailButton.cshtml:37]: Same as above. Create TODO or remove.

File: src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailButton.cshtml (hunk 1 above)
Comment (itm-mne) [inline src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailButton.cshtml:14]: Maybe add TODO, if section is to be used later. Else consider removing.

File: src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailButton.cshtml (hunk 2 above)
Comment (itm-mne) [inline src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailButton.cshtml:37]: Maybe add TODO, if section is to be used later. Else consider removing.

File: src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailLayout.cshtml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        div {margin:0 !important;}
+      </style>
+      <noscript>
+        <xml>
+          <o:OfficeDocumentSettings>
+            <o:PixelsPerInch>96</o:PixelsPerInch>
+          </o:OfficeDocumentSettings>
+        </xml>
+      </noscript>
+    <![endif]-->
```
Comment (itm-mne) [inline src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailLayout.cshtml:22]: Maybe add TODO, if section is to be used later. Else consider removing.

**Review by itm-mne — APPROVED** [review]
Consider adding TODO's to existing sections that are commented out, or consider removing the sections. Otherwise looks fine.

File: src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailLayout.cshtml (hunk 3 above)
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailLayout.cshtml:22]: Added a TODO comment.

File: src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailButton.cshtml (hunk 2 above)
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailButton.cshtml:37]: Added a TODO comment.

File: src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailButton.cshtml (hunk 1 above)
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailButton.cshtml:14]: Added a TODO comment.

File: src/Pelican.Application/Pelican.Application.csproj [hunk 4]
```diff
@@ -13,6 +13,7 @@
 		<PackageReference Include="GreenDonut" Version="12.15.2" />
 		<PackageReference Include="HotChocolate.Types" Version="12.15.2" />
 		<PackageReference Include="Microsoft.AspNetCore.Http" Version="2.2.2" />
+		<PackageReference Include="Microsoft.AspNetCore.Mvc.Razor" Version="2.2.0" />
```
Comment (Nicolai-Bak) [inline src/Pelican.Application/Pelican.Application.csproj:16]: Looks like this is only used in `Pelican.Infrastructure.RazorEmails` maybe it should be moved to that project

File: src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailButton.cshtml [hunk 5]
```diff
@@ -0,0 +1,40 @@
+﻿@using Pelican.Infrastructure.RazorEmails.Views.Shared
+
+@model EmailButtonViewModel
+
+<div align="center" class="button-container" style="padding-top:15px;padding-right:10px;padding-bottom:15px;padding-left:10px;">
+    <!-- TODO: Take a position on whether to keep or removed the commented code
```
Comment (Nicolai-Bak) [inline src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailButton.cshtml:6]: As it works without I think we should just remove the commented code

File: src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailLayout.cshtml [hunk 6]
```diff
@@ -0,0 +1,98 @@
+﻿<!DOCTYPE html>
+<html lang="en" xmlns="https://www.w3.org/1999/xhtml" xmlns:o="urn:schemas-microsoft-com:office:office">
+
+<head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width,initial-scale=1">
+    <meta name="x-apple-disable-message-reformatting">
+    <title></title>
+    <!-- 
+        TODO: Take a position on whether to keep or removed the commented code
```
Comment (Nicolai-Bak) [inline src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailLayout.cshtml:10]: Would prefer removing comments

File: src/Pelican.App/Pelican.App.csproj [hunk 7]
```diff
@@ -13,12 +13,14 @@
 			<PrivateAssets>all</PrivateAssets>
 			<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
 		</PackageReference>
+		<PackageReference Include="SendGrid" Version="9.28.1" />
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.App/Pelican.App.csproj:16]: Removed :)

File: src/Pelican.App/Program.cs [hunk 8]
```diff
@@ -2,18 +2,22 @@
...
 using Pelican.Infrastructure.Persistence;
+using Pelican.Infrastructure.RazorEmails;
 using Pelican.Presentation.Api;
 using Pelican.Presentation.GraphQL;
 
 const string CORS_PROD = "CorsProd";
 const string ALLOW_ALL = "AllowAll";
 
 var builder = WebApplication.CreateBuilder(args);
+builder.Services.AddRazorPages();
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.App/Program.cs:13]: It doesn't work in the .AddRazorEmails :/

File: src/Pelican.App/Program.cs [hunk 9]
```diff
@@ -2,18 +2,22 @@
...
 builder.Services.AddDomain(builder.Configuration, builder.Environment.IsProduction());
+builder.Services.AddRazorEmails();
 builder.Services.AddPersistince(builder.Configuration, builder.Environment.IsProduction());
 builder.Services.AddApplication(builder.Configuration);
 builder.Services.AddHubSpot();
 
+
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.App/Program.cs:20]: Removed newline

File: src/Pelican.App/appsettings.json [hunk 10]
```diff
@@ -42,5 +42,34 @@
...
+	},
+	"MailSettings": {
+		"Mail": "pelicanbirdthemedcrm@gmail.com",
+		"DisplayName": "PelicanSupport",
+		"Username": "apikey",
+		"Password": "PelicanSendgridSMTPRelayApiKey",
+		"Host": "smtp.sendgrid.net",
+		"Port": 587,
+		"BaseUrl": "https://pelican-linux-frontend-app.azurewebsites.net",
+		"DefaultEmails": [
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.App/appsettings.json:54]: That is for if nothing is stored in the Db, I think it is fine to have at the beginning and then an admin can change them :D

File: src/Pelican.Application/Pelican.Application.csproj (hunk 4 above)
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Application/Pelican.Application.csproj:16]: Done.

File: src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailButton.cshtml (hunk 5 above)
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailButton.cshtml:6]: Removed

File: src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailLayout.cshtml (hunk 6 above)
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Infrastructure.RazorEmails/Views/Shared/EmailLayout.cshtml:10]: Removed

**Review by Nicolai-Bak — APPROVED** [review]

**Pull request merged**